### PR TITLE
adding utils g2c_compare and g2c_degrib2

### PR DESCRIPTION
**Pull request by edwardhartnett**
Part of #405

### CODE REVIEW

**Comment by edwardhartnett** [comment]
OK, more testing can be added in future PRs, but this is a good start.

In this PR I add two utilities, g2c_compare and g2c_degrib2. I plan to use them in grib_util testing.

Tests are also added for the new utilities.

This PR is ready to merge.

File: utils/g2c_compare.c [hunk 1]
```diff
@@ -0,0 +1,91 @@
+/** @file
```
Comment (AlexanderRichert-NOAA) [inline utils/g2c_compare.c:1]: I think there's something wrong with this test-- If in the test directory I run `../utils/g2c_compare -v data/gdaswave.t00z.wcoast.0p16.f000.grib2 data/gdaswave.t00z.wcoast.0p16.f000.grib2` but change the name of the first file to one that doesn't exist, then it returns 202. But if I set the second file name to one that doesn't exist, it returns 0. Is that intentional?

Comment (edwardhartnett) [reply utils/g2c_compare.c:1]: No that's not expected.

I will add better error handling in the next iteration...

**Review by AlexanderRichert-NOAA — APPROVED** [review]
No further concerns beyond g2c_compare error handling issue

**Pull request merged**